src/winforms/toga_winforms/app.py

```diff
@@ -82,3 +82,6 @@ def main_loop(self):
 
     def exit(self):
         self.native.Exit()
+
+    def set_on_exit(self, value):
```

Comment:
The `set_on_exit()` method should be added to the toga_dummy backend (since it will be a required part of the App interface); this will then require empty implementations in all the backends.